File: backend/src/main/java/com/technicalescaperoom/backend/controller/admin/AdminAuthController.java
package com.technicalescaperoom.backend.controller.admin;

import com.technicalescaperoom.backend.dto.admin.AdminLoginRequest;
import com.technicalescaperoom.backend.service.admin.AdminAuthService;
import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/admin")
@RequiredArgsConstructor
public class AdminAuthController {

    public static final String ADMIN_COOKIE_NAME = "ADMIN_SESSION";
    private final AdminAuthService adminAuthService;

    @PostMapping("/login")
    public ResponseEntity<?> login(@Valid @RequestBody AdminLoginRequest request, HttpServletResponse response) {
        try {
            String token = adminAuthService.login(request.getPassword());

            Cookie cookie = new Cookie(ADMIN_COOKIE_NAME, token);
            cookie.setHttpOnly(true);
            cookie.setSecure(true);
            cookie.setPath("/");
            cookie.setMaxAge(24 * 60 * 60); // 24 hours
            cookie.setAttribute("SameSite", "None");
            response.addCookie(cookie);

            return ResponseEntity.ok(Map.of(
                    "message", "Login successful",
                    "token", token,
                    "sessionToken", token
            ));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(Map.of("error", e.getMessage()));
        }
    }
    
    @PostMapping("/logout")
    public ResponseEntity<?> logout(HttpServletResponse response) {
        Cookie cookie = new Cookie(ADMIN_COOKIE_NAME, null);
        cookie.setHttpOnly(true);
        cookie.setSecure(true);
        cookie.setPath("/");
        cookie.setMaxAge(0);
        cookie.setAttribute("SameSite", "None");
        response.addCookie(cookie);
        return ResponseEntity.ok(Map.of("message", "Logged out"));
    }
}

File: backend/controllers/admin/adminAuthController.js
const adminAuthService = require('../../services/admin/adminAuthService');
const { extractAdminToken } = require('../../middleware/authMiddleware');

class AdminAuthController {
  async login(req, res, next) {
    try {
      const { password } = req.body;
      if (!password) {
        return res.status(400).json({ error: 'Password is required' });
      }

      const token = await adminAuthService.login(password);

      res.cookie('ADMIN_SESSION', token, {
        httpOnly: true,
        secure: true,
        sameSite: 'none',
        maxAge: 24 * 60 * 60 * 1000,
        path: '/'
      });

      return res.json({
        message: 'Login successful',
        token,
        sessionToken: token
      });
    } catch (err) {
      return res.status(401).json({ error: err.message });
    }
  }

  async logout(req, res, next) {
    try {
      const token = extractAdminToken(req);
      if (token) {
        await adminAuthService.logout(token);
      }

      res.cookie('ADMIN_SESSION', '', {
        httpOnly: true,
        secure: true,
        sameSite: 'none',
        maxAge: 0,
        path: '/'
      });

      return res.json({ message: 'Logged out' });
    } catch (err) {
      next(err);
    }
  }
}

module.exports = new AdminAuthController();
